core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -189,6 +189,9 @@ UNWIND(	.cantunwind)
 	orr	r0, r0, #SCTLR_A
 	bic	r0, r0, #SCTLR_C
 	bic	r0, r0, #SCTLR_I
+#if defined(CFG_HWSUPP_MEM_PERM_WXN) && defined(CFG_CORE_RWDATA_NOEXEC)
```

Comment:
Some HW (C-A5, C-A9) do not support WXN and UWXN in SCTLR.
So we need this `CFG_HWSUPP_MEM_PERM_WXN`.
This is why the directive is not used in AARCH64, as all ARMv8-A do support SCTLR[WXN].